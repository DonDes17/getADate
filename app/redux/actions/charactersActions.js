import {
  GET_CHARACTERS,
  GET_CHARACTER_DETAILS,
  GET_MORE_PAGES,
} from '../constants';

// Action creator pour récupérer les personnages
export const getCharacters = () => ({ type: GET_CHARACTERS });

// Action creator pour récupérer une page supplémentaire
export const getMorePages = page => ({ type: GET_MORE_PAGES, page });

// Action creator pour récupérer les datails d'un personnage
export const getCharactersDetails = id => ({ type: GET_CHARACTER_DETAILS, id });
