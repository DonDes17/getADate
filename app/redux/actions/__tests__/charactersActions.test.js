import { getCharacters, getCharactersDetails } from '../charactersActions';


describe('CharactersActions', () => {
  it('checking formatted action', () => {
    // const expected = { type: 'GET_CHARACTERS' };
    // const actual = getCharacters();
    // expect(expected).toEqual(actual);
    expect(getCharacters()).toMatchSnapshot();
  });

  it('checking if the action is well formatted', () => {
    const id = 2;
    // const expected = { type: 'GET_CHARACTER_DETAILS', id };
    // const actual = getCharactersDetails(id);
    // expect(expected).toEqual(actual);
    expect(getCharactersDetails(id)).toMatchSnapshot();
  });
});
