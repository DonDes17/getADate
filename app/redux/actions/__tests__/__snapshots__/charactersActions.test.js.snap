// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersActions checking formatted action 1`] = `
Object {
  "type": "GET_CHARACTERS",
}
`;

exports[`CharactersActions checking if the action is well formatted 1`] = `
Object {
  "id": 2,
  "type": "GET_CHARACTER_DETAILS",
}
`;
